controller/src/controller/storage/kubernetes/fsadmin.py

```diff
@@ -0,0 +1,152 @@
+        logger: BoundLogger,
+    ) -> None:
+        self._config = config
+        self._logger = logger
+        self._metadata = metadata_storage
+        self._pod = PodStorage(api_client, logger)
+        self._pvc = PersistentVolumeClaimStorage(api_client, logger)
+
+    async def create(self, objects: FSAdminObjects, timeout: Timeout) -> None:
+        """Create all of the Kubernetes objects for an fsadmin instance.
+
+        Create the objects in Kubernetes and then wait for the fsadmin pod
+        to start.
+
+        Parameters
+        ----------
+        objects
+            Kubernetes objects making up the fsadmin environment.
+        timeout
+            How long to wait for the fsadmin pod to start.
+
+        Raises
+        ------
+        KubernetesError
+            Raised if there is some failure in a Kubernetes API call.
+        TimeoutError
+            Raised if fsadmin is not ready before the provided timeout expires.
+
+        Notes
+        -----
+        This is conceptually similar to the fileserver, but since it's a
+        cluster-wide singleton, if the objects all exist, that's fine.  We
+        don't try to delete and recreate, but use the existing fsadmin
+        instance.
+        """
+        ns = self._metadata.namespace
+        for pvc in objects.pvcs:
+            pvcobj = await self._pvc.read(pvc.metadata.name, ns, timeout)
+            if pvcobj is None:
+                await self._pvc.create(ns, pvc, timeout)
+
+        pod = objects.pod
+        existing_pod = await self._pod.read(pod.metadata.name, ns, timeout)
+        if existing_pod is None:
+            await self._pod.create(ns, pod, timeout)
+
+        # Wait for the pod to start.
+        await self._pod.wait_for_phase(
+            pod.metadata.name,
+            ns,
+            until_not={PodPhase.UNKNOWN, PodPhase.PENDING},
+            timeout=timeout,
+        )
+
+    async def delete(self, objects: FSAdminObjects, timeout: Timeout) -> None:
+        """Delete the fsadmin instance.
+
+        Parameters
+        ----------
+        objects
+            Kubernetes objects making up the fsadmin environment.
+        timeout
+            Timeout on operation.
+
+        Raises
+        ------
+        KubernetesError
+            Raised if there is some failure in a Kubernetes API call.
+        TimeoutError
+            Raised if fsadmin namespace is not deleted within provided timeout.
+        """
+        ns = self._metadata.namespace
+        pod = objects.pod
+        pvcs = objects.pvcs
+        existing_pod = await self._pod.read(pod.metadata.name, ns, timeout)
+        if existing_pod is not None:
+            await self._pod.delete(
+                existing_pod.metadata.name, ns, timeout, wait=True
+            )
+        for pvc in pvcs:
+            if await self._pvc.read(pvc.metadata.name, ns, timeout) is None:
+                continue
+            await self._pvc.delete(pvc.metadata.name, ns, timeout)
```

Comment:
Just delete Kubernetes resources, don't check if they exist first. This is all handled by `KubernetesObjectDeleter` already by catching the 404 error from Kubernetes.